src/openrct2-ui/windows/Changelog.cpp

```diff
@@ -41,273 +41,300 @@ static constexpr const rct_string_id WINDOW_TITLE = STR_CHANGELOG_TITLE;
-    return std::string((std::istreambuf_iterator<char>(fs)), std::istreambuf_iterator<char>());
-}
 
-static bool window_changelog_read_file()
-{
-    std::string _changelogText;
-    try
+    /**
+     * @brief Get the absolute path for the changelog file as a std::string
+     *
+     * @return std::string
+     */
+    std::string GetChangelogPath()
     {
-        _changelogText = GetChangelogText();
+        auto env = GetContext()->GetPlatformEnvironment();
+        return env->GetFilePath(PATHID::CHANGELOG);
     }
-    catch (const std::bad_alloc&)
+
+    /**
+     * @brief Attempts to read the changelog file, returns true on success
+     *
+     * @return true
+     * @return false
+     */
+    bool ReadChangelogFile()
     {
-        log_error("Unable to allocate memory for changelog.txt");
-        return false;
+        std::string _changelogText;
+        try
+        {
+            _changelogText = GetChangelogText();
+        }
+        catch (const std::bad_alloc&)
+        {
+            log_error("Unable to allocate memory for changelog.txt");
+            return false;
+        }
+        catch (const std::exception&)
+        {
+            log_error("Unable to read changelog.txt");
+            return false;
+        }
+
+        ProcessChangelogText(_changelogText);
+        return true;
     }
-    catch (const std::exception&)
+
+    /**
+     * @brief Pushes a line of text to the changelog
+     *
+     * @param text
+     */
+    void ProcessChangelogText(const std::string& text)
     {
-        log_error("Unable to read changelog.txt");
-        return false;
+        std::string::size_type pos = 0;
+        std::string::size_type prev = 0;
+        while ((pos = text.find("\n", prev)) != std::string::npos)
+        {
+            _changelogLines.push_back(text.substr(prev, pos - prev));
+            prev = pos + 1;
+        }
+
+        // To get the last substring (or only, if delimiter is not found)
+        _changelogLines.push_back(text.substr(prev));
+
+        _changelogLongestLineWidth = 0;
+        for (const auto& line : _changelogLines)
+        {
+            int32_t linewidth = gfx_get_string_width(line.c_str(), FontSpriteBase::MEDIUM);
+            _changelogLongestLineWidth = std::max(linewidth, _changelogLongestLineWidth);
+        }
     }
+};
 
-    window_changelog_process_changelog_text(_changelogText);
-    return true;
+rct_window* window_changelog_open(int personality)
+{
+    auto* window = window_bring_to_front_by_class(WC_CHANGELOG);
+    if (window == nullptr)
+    {
+        // Create a new centered window
+        int32_t screenWidth = context_get_width();
+        int32_t screenHeight = context_get_height();
+        int32_t width = (screenWidth * 4) / 5;
+        int32_t height = (screenHeight * 4) / 5;
+
+        auto pos = ChangelogWindow::GetCentrePositionForNewWindow(width, height);
+        window = WindowCreate<ChangelogWindow>(WC_CHANGELOG, pos, width, height, WF_RESIZABLE);
```

Comment:
You can eliminate the static cast below by introducing a new variable here
```suggestion
        auto* newWindow = WindowCreate<ChangelogWindow>(WC_CHANGELOG, pos, width, height, WF_RESIZABLE);
```